jose-jwt/compression/DeflateCompression.cs

```diff
@@ -28,24 +28,22 @@ public byte[] Compress(byte[] plainText)
 
         public byte[] Decompress(byte[] compressedText)
         {
-            byte[] buffer = new byte[maxBufferSizeBytes];
-
             try
             {
-                using (MemoryStream ms = new MemoryStream(buffer))
+                using (MemoryStream ms = new CappedMemoryStream(maxBufferSizeBytes))
                 {
                     using (MemoryStream compressedStream = new MemoryStream(compressedText))
                     {
                         using (DeflateStream deflater = new DeflateStream(compressedStream, CompressionMode.Decompress))
                         {
                             deflater.CopyTo(ms);
                         }
-                    }
-
-                    return Arrays.Truncate(ms.ToArray(), ms.Position);
+                    }
+
+                    return ms.ToArray();
                 }
             }
-            catch(NotSupportedException e)
+            catch (NotSupportedException e)
```

Comment:
may be we can now replace to some better exception? Like `JoseException` of subclass ?